### ISSUE TITLE
Can't get connection established

### ISSUE THREAD

**Issue by DMaddison** [Knowinnovation/bluetime#41]
Behavior:  Start up kitimer on iPad2 and iPhone5.  Touch on + button on iPad, say, and invite the iPhone.  The iPad then looks good, which a check mark beside the + button.  But the iPhone just sits there with "Connecting" and the spinning thing for many minutes.  The check mark does appear beside the iPhone's + button, but "Connecting" never goes away.  Curious thing is that they ARE connected at least to some extent, as if press the Start button on the iPad, the timer starts going on the iPhone, behind the "Connecting" notice.    Because the Connecting notice never goes away, you can't actually use the link.

iOS 9.3.5 (13G36) on both machines

They are about 5 cm apart.

I've tried this with WiFi and Bluetooth turned on, and with just WiFi, and just Bluetooth.

**Comment by DMaddison** [comment]
I should have noted that I rebooted both machines and tried again; same result.

**Comment by andyburnett** [comment]
Just tested this on various devices. I think the problem was introduced in 9.5.3.

I have just tested with. 10.x device connecting to a 9.5.3 device and it all works correctly.

**Comment by andyburnett** [comment]
Results of more testing

Bluetooth connections seem to be completely broken 10 -> 10 and 10 -> 9.x

Wifi connections do work 10 -> 10 and 10 -> 9.x

9.x -> 10 and 9.x -> 9.x are completely  unable to establish a connection.

**Comment by doctor3w** [comment]
Makes sense, I was expecting some issues when iOS 10 rolled out. I'll try and get that fixed and update the App Store app so it's compatible.

**Comment by andyburnett** [comment]
Great

**Comment by DMaddison** [comment]
Thanks for looking into this!   I can't get it to work in either direction with an iPhone with 10.0.1 and an iPad with 9.3.5 (latest version) with WiFi (or Bluetooth for that matter).  

Drew: note that this is also broken fully within the 9.3.5 world - i.e., it is not just an iOS 10 issue.

**Comment by andyburnett** [comment]
Hmm,
We need to do more testing on this. We have it working at the event we are running right now, but this must be some strange combination of iOS and devices.

**Comment by dsl101** [comment]
I have a phone and ipad both running 9.3.5, and I also see the stuck 'connecting' message on the ipad after accepting an invitation from the phone, or vice versa, on both wifi and BT.

The only solution I found was to set 'auto accept invitations' - then I was able to get past the 'connecting' message. @DMaddison that might be a short term workaround.

@drewsdunne let me know if you want anything testing specifically.

**Comment by andyburnett** [comment]
@DMaddison Drew has just pushed a new version of the code, and this now seems to work correctly across ios 10 and ios 9.  If you have a moment, can you give it a try?  Please note, you need to set your target device to auto accept connection requests (we found a bug about this)
